streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamsPartitionAssignor.java

```diff
@@ -473,13 +482,22 @@ public GroupAssignment assign(final Cluster metadata, final GroupSubscription gr
                                 // if this topic is one of the sink topics of this topology,
                                 // use the maximum of all its source topic partitions as the number of partitions
                                 for (final String sourceTopicName : otherTopicsInfo.sourceTopics) {
-                                    final Integer numPartitionsCandidate;
+                                    final int numPartitionsCandidate;
                                     // It is possible the sourceTopic is another internal topic, i.e,
                                     // map().join().join(map())
                                     if (repartitionTopicMetadata.containsKey(sourceTopicName)) {
-                                        numPartitionsCandidate = repartitionTopicMetadata.get(sourceTopicName).numberOfPartitions();
+                                        numPartitionsCandidate =
+                                            repartitionTopicMetadata.get(sourceTopicName).numberOfPartitions();
                                     } else {
-                                        numPartitionsCandidate = metadata.partitionCountForTopic(sourceTopicName);
+                                        final Integer count = metadata.partitionCountForTopic(sourceTopicName);
+                                        if (count == null) {
```

Comment:
avoiding an unboxing warning.

Reply:
The topic may not be in the metadata due to some edge conditions, we cannot throw the exception in this case.

Reply:
This exception would always have happened anyway (it would have thrown an NPE on unboxing). What alternative handling would you recommend?

Reply:
Hmm, in that case we can do this as-is then: I need to think through what's possible under the context of KIP-429 anyways so we can have a thorough discussion in a later PR.